packages/liveblocks-react-lexical/src/comments/threads-panel.tsx

```diff
@@ -0,0 +1,429 @@
+
+    // Iterate over each thread and calculate its new position by taking into account the position of the previously positioned threads
+    for (const { thread, rect } of ascending) {
+      let top = rect.top - container.getBoundingClientRect().top;
+
+      for (const [id, position] of newPositions) {
+        // Retrieve the element associated with the thread
+        const el = elements.get(id);
+        if (el === undefined) continue;
+
+        if (
+          top >= position &&
+          top <= position + el.getBoundingClientRect().height
+        ) {
+          top = position + el.getBoundingClientRect().height;
+        }
+      }
+
+      newPositions.set(thread.id, top);
+    }
+
+    for (const { thread, rect } of descending.reverse()) {
+      // Retrieve the element associated with the current thread
+      const el = elements.get(thread.id);
+      if (el === undefined) continue;
+
+      let top = rect.top - container.getBoundingClientRect().top;
+      for (const [, position] of newPositions) {
+        if (top >= position - el.getBoundingClientRect().height) {
+          top = position - el.getBoundingClientRect().height;
+        }
+      }
+
+      newPositions.set(thread.id, top);
+    }
+
+    setPositions(newPositions);
+  }, [editor, threads, nodes, activeThreads, elements]);
+
+  useLayoutEffect(() => {
+    handlePositionThreads();
+  }, [handlePositionThreads]);
+
+  useEffect(() => {
+    return editor.registerUpdateListener(() => {
+      handlePositionThreads();
+    });
+  }, [editor, handlePositionThreads]);
+
+  useEffect(() => {
+    const observer = new ResizeObserver(handlePositionThreads);
+    for (const element of elements.values()) {
+      observer.observe(element);
+    }
+
+    return () => observer.disconnect();
+  }, [elements, handlePositionThreads]);
+
+  const root = useRootElement();
+
+  useEffect(() => {
+    if (root === null) return;
+    const observer = new ResizeObserver(handlePositionThreads);
+
+    observer.observe(root);
+    return () => observer.disconnect();
+  }, [root, handlePositionThreads]);
+
+  if (orderedThreads.length === 0) return;
+
+  return (
+    <div
+      {...divProps}
+      className={classNames(className, "lb-root lb-lexical-thread-panel")}
+      ref={containerRef}
+      style={{
+        position: "relative",
+        ...style,
+      }}
+    >
+      {orderedThreads.map(({ thread, rect }) => {
+        let top = rect.top;
+
+        if (positions.has(thread.id)) {
+          top = positions.get(thread.id)!;
+        }
+
+        const isActive = activeThreads.includes(thread.id);
+
+        return (
+          <ThreadWrapper
+            key={thread.id}
+            Thread={Thread}
+            thread={thread}
+            onItemAdd={onItemAdd}
+            onItemRemove={onItemRemove}
+            style={{
+              position: "absolute",
+              transform: `translateX(${isActive ? -12 : 0}px) translateY(${top}px)`,
```

Comment:
@marcbouchenoire Threads that are active will also be translated left by `-12px`. Any suggestion on this number. I think it is okay to hard code this for now and in future, if users would like a different value, we could allow a prop to override this.

Reply:
I think we can make it a CSS variable defined for this component only.

I also noticed that you don't actually use the `gap` value in your layout calculations so I'm tempted to also make it a CSS variable.

Reply:
I personally generally prefer CSS variables because you can set them in React and in CSS (supporting responsive values, etc), compared to props where you're stuck in React only (it might be fine for many use-cases but it removes possibilities).

Maybe we could have a mixed API with a prop that sets the default value of a CSS variable, best of both worlds?